cmd/thanos/config.go

```diff
@@ -0,0 +1,61 @@
+	tlsSrvCert     string         `flagarize:"name=grpc-server-tls-cert|help=TLS Certificate for gRPC server, leave blank to disable TLS|default="`
+	tlsSrvKey      string         `flagarize:"name=grpc-server-tls-key|help=TLS Key for the gRPC server, leave blank to disable TLS|default="`
+	tlsSrvClientCA string         `flagarize:"name=grpc-server-tls-client-ca|help=TLS CA to verify clients against. If no client CA is specified, there is no client verification on server side. (tls.NoClientCert)|default="`
+}
+
+type httpConfig struct {
+	bindAddress string         `flagarize:"name=http-address|help=Listen host:port for HTTP endpoints.|default=0.0.0.0:10902"`
+	gracePeriod model.Duration `flagarize:"name=http-grace-period|help=Time to wait after an interrupt received for HTTP Server.|default=2m"`
+}
+
+type prometheusConfig struct {
+	url          url.URL       `flagarize:"name=prometheus.url|help=URL at which to reach Prometheus's API. For better performance use local network.|default=http://localhost:9090"`
+	readyTimeout time.Duration `flagarize:"name=prometheus.ready_timeout|help=Maximum time to wait for the Prometheus instance to start up|default=10m"`
+}
+
+type connectionConfig struct {
+	maxIdleConns        int `flagarize:"name=receive.connection-pool-size|help=Controls the http MaxIdleConns. Default is unlimited|default=0"`
+	maxIdleConnsPerHost int `flagarize:"name=receive.connection-pool-size-per-host|help=Controls the http MaxIdleConnsPerHost|default=100"`
+}
+
+type tsdbConfig struct {
+	path string `flagarize:"name=tsdb.path|help=Data directory of TSDB.|default=./data"`
+}
+
+type reloaderConfig struct {
+	confFile        string   `flagarize:"name=reloader.config-file|help=Config file watched by the reloader."`
+	envVarConfFile  string   `flagarize:"name=reloader.config-envsubst-file|help=Output file for environment variable substituted config file."`
+	ruleDirectories []string `flagarize:"name=reloader.rule-dir|help=Rule directories for the reloader to refresh (repeated field)."`
+}
+
+type shipperConfig struct {
+	uploadCompacted *bool `flagarize:"name=shipper.upload-compacted|help=If true sidecar will try to upload compacted blocks as well. Useful for migration purposes. Works only if compaction is disabled on Prometheus. Do it once and then disable the flag when done.|default=false"`
+	ignoreBlockSize *bool `flagarize:"name=shipper.ignore-unequal-block-size|help=If true sidecar will not require prometheus min and max block size flags to be set to the same value. Only use this if you want to keep long retention and compaction enabled on your Prometheus instance, as in the worst case it can result in ~2h data loss for your Thanos bucket storage.|default=false"`
+}
+
+type sidecarComponentConfig struct {
+	grpc       grpcConfig
+	http       httpConfig
+	prometheus prometheusConfig
+	tsdb       tsdbConfig
+	connection connectionConfig
+	reloader   reloaderConfig
+	shipper    shipperConfig
+	objStore   *extflag.PathOrContent
```

Comment:
This needs some work or we should use `flagarize` PathOrContent instead (and remove our flag)